components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/dao/impl/UserSessionDAOImpl.java

```diff
@@ -89,4 +90,34 @@ public UserSession getSession(String sessionId) throws SessionManagementServerEx
         }
         return null;
     }
+
+    /**
+     * Get federated authentication session mapping info during the federated idp init logout
+     *
+     * @param fedIdpSessionId
+     * @return A FederatedUserSession containing federated authentication session details
+     * @throws SessionManagementServerException
+     */
+    public FederatedUserSession getFederatedAuthSessionDetails(String fedIdpSessionId)
+            throws SessionManagementServerException {
+
+        FederatedUserSession federatedUserSession;
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        try {
+            federatedUserSession = jdbcTemplate
+                    .fetchSingleRecord(SQLQueries.SQL_GET_FEDERATED_AUTH_SESSION_INFO_BY_SESSION_ID,
+                            (resultSet, rowNumber) -> new FederatedUserSession(
+                                    resultSet.getString(1),
```

Comment:
Shall we get the columns by name here to improve readability

Reply:
Changed in 3ae4174